src/cluster_migrateslots.c

```diff
@@ -1231,34 +1233,128 @@ int childSnapshotForSyncSlot(int req, rio *rdb, void *privdata) {
             slotRange *r = (slotRange *)ln->value;
             for (int slot = r->start_slot; slot <= r->end_slot; slot++) {
                 if (rewriteSlotToAppendOnlyFileRio(
-                        rdb, db_num, slot, &key_count) == C_ERR) return C_ERR;
+                        aof, db_num, slot, &key_count) == C_ERR) return C_ERR;
             }
         }
     }
-    rioWrite(rdb, "*3\r\n", 4);
-    rioWriteBulkString(rdb, "CLUSTER", 7);
-    rioWriteBulkString(rdb, "SYNCSLOTS", 9);
-    rioWriteBulkString(rdb, "SNAPSHOT-EOF", 12);
+    rioWrite(aof, "*3\r\n", 4);
+    rioWriteBulkString(aof, "CLUSTER", 7);
+    rioWriteBulkString(aof, "SYNCSLOTS", 9);
+    rioWriteBulkString(aof, "SNAPSHOT-EOF", 12);
     return C_OK;
 }
 
+void killSlotMigrationChild(void) {
+    int statloc;
+    /* No slot migration child? return. */
+    if (server.child_type != CHILD_TYPE_SLOT_MIGRATION) return;
+    /* Kill slot migration child, wait for child exit. */
+    serverLog(LL_NOTICE, "Killing running slot migration child: %ld", (long)server.child_pid);
+    if (kill(server.child_pid, SIGUSR1) != -1) {
+        while (waitpid(-1, &statloc, 0) != server.child_pid);
+    }
+    resetChildState();
+    serverLog(LL_NOTICE, "Slot migration child %ld killed", (long)server.child_pid);
```

Comment:
```suggestion
    serverLog(LL_NOTICE, "Killing running slot migration child: %ld", (long)server.child_pid);
    if (kill(server.child_pid, SIGUSR1) != -1) {
        while (waitpid(-1, &statloc, 0) != server.child_pid);
    }
    serverLog(LL_NOTICE, "Slot migration child %ld killed", (long)server.child_pid);
    resetChildState();
```
This should fix the log:

```
20601:M 23 Aug 2025 08:23:48.111 * Slot migration child -1 killed
```